### PR TITLE
single waveformwidget class for all allshader waveform widget types

### PR DESCRIPTION
**Pull request by m0dB**
Use a single allshader::WaveformWidget class for all the allshader Waveform widget types.

Note how I automatically deal with adding the slip renderer layers.

I also did some cleanup in WaveformWidgetFactory, some related with / required by this change to use a single allshader::WaveformWidget. I also introduced a struct WaveformWidgetVars, to be used instead of having to add multiple static functions (static bool useGL, etc), but I only use this for the allshader::WaveformWidget for now.

### CODE REVIEW

File: src/waveform/waveformwidgetfactory.h [hunk 1]
```diff
@@ -231,9 +232,16 @@ class WaveformWidgetFactory : public QObject, public Singleton<WaveformWidgetFac
     void renderSelf();
     void swapSelf();
 
+    void addHandle(
+            QHash<WaveformWidgetType::Type, QList<WaveformWidgetBackend>>&
+                    collectedHandles,
```
Comment (Swiftb0y) [inline src/waveform/waveformwidgetfactory.h:237]: I don't like the out-parameter here, but we can clean that up some time else.

**Review by Swiftb0y — COMMENTED** [review]
small syntax nitpick, rest is good enough. Thank you.

**Comment by ywwg** [comment]
@acolombier up to you to review / merge :)

File: src/waveform/waveformwidgetfactory.h (hunk 1 above)
Comment (m0dB) [reply src/waveform/waveformwidgetfactory.h:237]: I agree, I was thinking to put all this in a class with the QHash as a member. I'll add a TODO so we don't forget about it.

Comment (acolombier) [reply src/waveform/waveformwidgetfactory.h:237]: Just referencing [mixxxdj/mixxx#13245](https://github.com/mixxxdj/mixxx/issues/13245) for tracking

**Comment by m0dB** [comment]
@acolombier I think you can merge this already. I will create a follow-up PR with the suggestions from @Swiftb0y when I have a moment.

**Review by acolombier — APPROVED** [review]

**Pull request merged**